modules/axhal/src/arch/x86_64/syscall.rs

```diff
@@ -17,6 +17,7 @@ core::arch::global_asm!(
 
 pub(super) fn handle_syscall(tf: &mut TrapFrame) {
     tf.rax = crate::trap::handle_syscall(tf, tf.rax as usize) as u64;
+    crate::trap::post_trap_callback(tf, true);
```

Comment:
Move this to `x86_syscall_handler`, after `handle_syscall` and before `switch_to_user_fs_base`